blog/_posts/2021-07-06-distributed-rdataframe.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "RDataFrame is going distributed!"
+layout: archive
+author: Vincenzo Eduardo Padulano
+---
+
+A recently introduced Python package enables distributing ROOT RDataFrame workloads to a set of remote resources.
+This feature is available in experimental phase since the latest ROOT 6.24 release, allowing users to write and run
+their applications from within the same interface while steering the computations to an Apache Spark cluster.
+
+## In detail
+
+[RDataFrame](https://root.cern/doc/master/classROOT_1_1RDataFrame.html) is ROOT's high-level interface for data
+analysis and since its release in 6.14 has seen tons of usages in real world analysis use cases and other examples in
+the wild. Parallelism has always been a staple of its design with support for executing the event loop on all cores
+of a machine thanks to implicit multi-threading. Since ROOT 6.24, this aspect of RDataFrame has been enhanced further
+with distributed computing capabilities, allowing users to run their analysis on multi-node computing clusters through
+widely used frameworks such as [Apache Spark](https://spark.apache.org/), [Dask](https://dask.org/) or
+[AWS Lambda](https://aws.amazon.com/lambda/). Here is an example usage of the tool (requires the Python `pyspark`
+package):
+
+```python
+import ROOT
+ 
+# Point RDataFrame calls to the Spark specific RDataFrame
+RDataFrame = ROOT.RDF.Experimental.Distributed.Spark.RDataFrame
+ 
+# It still accepts the same constructor arguments as traditional RDataFrame
+df = RDataFrame("mytree", "myfile.root")
+ 
+# Continue the application with the traditional RDataFrame API
+sum = df.Filter("x > 10").Sum("y")
+h = df.Histo1D("x")
+ 
+print(sum.GetValue())
+h.Draw()
+```
+
+The main goal is to support distributed execution of any RDataFrame application. This has led to the creation of a 
+**Python** package that connects the RDataFrame API (available in Python through PyROOT) and the APIs of
+distributed computing frameworks, which are offered in Python in the vast majority of cases. Another key goal is to be
+able to support as many backends as possible, in order to avoid being tied to any particular infrastructure. This is
+achieved through a modular implementation that defines a generic task (representing the RDataFrame computation graph) to
+be executed on data. The input dataset is logically split into many ranges of entries, which will be sent to the
+distributed nodes for processing. Each range will then be paired to the generic task and submitted to the computing
+framework via a specific backend implementation. An added benefit of using RDataFrame is that the distributed
+tasks run C++ computations. This is made possible by PyROOT and cling. Currently the package offers support for running
+the application on a Spark cluster, but the package design will make it possible to add many more backends in the future.
```

Comment:
This sounds a bit repetitive to me w.r.t. what was written above (it's the second time the three backends are mentioned). Perhaps remove this and include the info about what is there (Spark) and what will come (Dask, AWS) in the first paragraph where you mention them?

Reply:
I actually made the post explicitly repetitive :smile: I think it's not easy to grasp all the meaning of this passage in a single sentence, plus the first paragraph it's a sort of "introduction" and just serves to mention the three backends currently "under our grasp", while more info about them is described in the next paragraphs.

Reply:
I can of course try to avoid repetitions, let me come up with something and I'll get back to you